onnxruntime/core/framework/session_state_utils.cc

```diff
@@ -307,9 +325,13 @@ common::Status SaveInitializedTensors(
       bool use_device_allocator_for_initializers =
           session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsUseDeviceAllocatorForInitializers, "0") == "1";
 
+      Tensor* p_tensor = buffered_tensors.find(name) != buffered_tensors.end()
```

Comment:
Why do you need to call release? So you want to keep the key-value pair there, but the next lookup should return a nullptr?

Reply:
Inside `DeserializeTensorProto`, a deleter is created. The deleter manages the lifetime of the tensor buffer, and it will release the tensor once the tensor is not needed any more. Therefore, the `unique_ptr` should give up the ownership.

Reply:
Then why don't you remove the key-value pair from the map at here?

Reply:
Is this line the only place that lookup the map?

Reply:
yes this is the only place that look up the map. I can remove the kv pair here after transfer ownership, but it does not make much difference. The destructor of the session state -> map will clean up the remaining pairs in the map.

Reply:
updated